scripts/base/init-bare.zeek

```diff
@@ -2983,6 +2983,14 @@ export {
 	##
 	## .. zeek:see:: smb_pipe_connect_heuristic
 	const SMB::pipe_filenames: set[string] &redef;
+
+	## The maximum number of messages for which to retain state
+	## about offsets, fids, or tree ids before discarding these
+	## to free up internal parser state. When state is discarded,
+	## :zeek:see:`smb2_discarded_messages_state` is raised.
```

Comment:
> before discarding these to free up internal parser state

Is this true if `enable_clear_state` is set to false? If that's the case, `smb2_discarded_messages_state` just reports the weird and exits.

Reply:
> Is this true if enable_clear_state is set to false? If that's the case, smb2_discarded_messages_state just reports the weird and exits.

The "internal parser state" (those std::maps in binpac) is cleared even when enable_clear_state=F. That setting only toggles script-layer state resetting. I see now that this is confusing. Didn't have `enable_clear_state` initially - I'll stuff a `_script_state` into the name, though not sure that's immediately better.